kernel/aster-nix/src/net/iface/common.rs

```diff
@@ -147,56 +149,93 @@ impl IfaceCommon {
 
     pub(super) fn poll<D: Device + ?Sized>(&self, device: &mut D) {
         let mut interface = self.interface.lock_irq_disabled();
+        let mut sockets = self.sockets.lock_irq_disabled();
+
         let timestamp = get_network_timestamp();
-        let has_events = {
-            let mut sockets = self.sockets.lock_irq_disabled();
-            interface.poll(timestamp, device, &mut sockets)
-            // drop sockets here to avoid deadlock
-        };
-        if has_events {
-            self.bound_sockets.read().iter().for_each(|bound_socket| {
-                if let Some(bound_socket) = bound_socket.upgrade() {
-                    bound_socket.on_iface_events();
+        let (has_events, poll_at) = {
+            let mut has_events = false;
+            let mut poll_at;
+            loop {
+                has_events |= interface.poll(timestamp, device, &mut sockets);
+                poll_at = interface.poll_at(timestamp, &sockets);
+                let Some(instant) = poll_at else {
+                    break;
+                };
+                if instant > timestamp {
+                    break;
                 }
```

Comment:
Do we really need this loop? It seems if `poll_at` is _not_ `None`, `instant` will always be greater than `timestamp`?

Reply:
Good catch!